src/LibVLCSharp.Forms/Shared/PlaybackControls.xaml.cs

```diff
@@ -821,14 +813,113 @@ private void OnAudioTracksChanged(object sender, EventArgs e)
+            {
+                AudioTracksListView.ItemsSource = audioTrackSouces;
+                VideoTracksListView.ItemsSource = videoTrackSouces;
+                SubtitlesTracksListView.ItemsSource = subtitleTracksSources;
+            }
+
+            if (TracksOverlayView != null)
+                TracksOverlayView.IsVisible = true;
+        }
+
+        private void AudioTracksItemTapped(object sender, ItemTappedEventArgs e)
+        {
+            var track = (TrackViewModel)e.Item;
+            var manager = Manager.Get<AudioTracksManager>();
+            SelectTrack(manager, track, AudioTracksListView);
+        }
+
+        private void VideoTracksItemTapped(object sender, ItemTappedEventArgs e)
+        {
+            var track = (TrackViewModel)e.Item;
+            var manager = Manager.Get<VideoTracksManager>();
+            SelectTrack(manager, track, VideoTracksListView);
+        }
+
+        private void SubtitlesTracksItemTapped(object sender, ItemTappedEventArgs e)
+        {
+            var track = (TrackViewModel)e.Item;
+            var manager = Manager.Get<SubtitlesTracksManager>();
+            SelectTrack(manager, track, SubtitlesTracksListView);
+        }
+
+        private void SelectTrack(TracksManager manager, TrackViewModel track, ListView? tracksListview)
+        {
+            var tracks = manager.Tracks;
+            var currentTrackId = manager.CurrentTrackId;
+            if (tracks == null || track.Selected)
+                return;
+
+            var foundTrack = tracks.FirstOrDefault(t => t.Name == track.Name);
+            if (foundTrack.Name != null)
+            {
+                manager.CurrentTrackId = foundTrack.Id;
+                UpdateTracksListviewItemsSource(track, tracksListview);
+            }
+            else
+            {
+                manager.CurrentTrackId = -1;
+            }
+        }
+
+        private void UpdateTracksListviewItemsSource(TrackViewModel selectedTrack, ListView? trackListView)
+        {
+            if (trackListView != null)
+            {
+                var itemSources = (ObservableCollection<TrackViewModel>)trackListView.ItemsSource;
+                var previousTrack = itemSources.FirstOrDefault(t => t.Selected);
+
+                itemSources.ElementAt(previousTrack.Id).Selected = false;
+                itemSources.ElementAt(selectedTrack.Id).Selected = true;
+            }
+        }
+
+        private ObservableCollection<TrackViewModel> LoadTracks(TracksManager manager)
         {
-            await SelectTrackAsync(Manager.Get<AudioTracksManager>(), ResourceManager.GetString(nameof(Strings.AudioTracks)));
+            var allTracks = new ObservableCollection<TrackViewModel>();
+            var tracks = manager.Tracks;
+            try
+            {
+                var currentTrackId = manager.CurrentTrackId;
+                var index = -1;
+                if (tracks != null)
```

Comment:
This should be tested right after the `var tracks = manager.Tracks;` line (return immediately in case it's null.